vertical-pod-autoscaler/pkg/updater/logic/updater.go

```diff
@@ -177,11 +177,29 @@ func (u *updater) RunOnce(ctx context.Context) {
 	}
 	timer.ObserveStep("AdmissionInit")
 
+	// wrappers for metrics which are computed every loop run
+	controlledPodsCounter := metrics_updater.NewControlledPodsCounter()
+	evictablePodsCounter := metrics_updater.NewEvictablePodsCounter()
+	vpasWithEvictablePodsCounter := metrics_updater.NewVpasWithEvictablePodsCounter()
+	vpasWithEvictedPodsCounter := metrics_updater.NewVpasWithEvictedPodsCounter()
+
+	// using defer to protect against 'return' after evictionRateLimiter.Wait
+	defer controlledPodsCounter.Observe()
+	defer evictablePodsCounter.Observe()
+	defer vpasWithEvictablePodsCounter.Observe()
+	defer vpasWithEvictedPodsCounter.Observe()
+
 	for vpa, livePods := range controlledPods {
+		vpaSize := len(livePods)
+		controlledPodsCounter.Add(vpaSize, vpaSize)
```

Comment:
nit: I think controlledPods may include those with VPA in "off" or "initial", which makes them "controlled by VPA" not necessarily "controlled by updater" (as in description of the metric).

Reply:
If I understand the code correctly we only take VPAs in auto/recreate mode into account:
1. we filter such VPA objects to create `vpas` variable, lines 130+
2. we use `vpas` to filter controlled Pods, lines 166+

I added a comment for the future readers.

Reply:
Thanks, for some reason I thought this was unfiltered